backend/common/ai_drive.py

```diff
@@ -210,31 +209,40 @@ def ml_drive(
         problem_type (str): "classification" or "regression" problem
         target (str, optional): name of target column. Defaults to None.
         features (list, optional): list of columns in dataframe for the feature based on user selection. Defaults to None.
-        default (bool, optional): use the iris dataset for default classifiction or california housing for default regression. Defaults to False.
+        default (bool, optional): use the iris dataset for default classification or california housing for default regression. Defaults to False.
         test_size (float, optional): size of test set in train/test split (percentage). Defaults to 0.2.
         shuffle (bool, optional): should the dataset be shuffled prior to train/test split
     """
-    try:
-        if not default:
-            if fileURL:
-                read_dataset(fileURL)
-            elif json_csv_data_str:
-                pass
-            else:
-                raise ValueError("Need a file input")
+    params = trainspace_data.parameters_data
+
+    target = params.get("target_col", None)
+    features = params.get("features", None)
+    problem_type = params["problem_type"]
+    default = (
+        trainspace_data.dataset_data["name"]
+        if trainspace_data.dataset_data.get("is_default_dataset")
+        else None
+    )
+    shuffle = params.get("shuffle", True)
+    test_size = params.get("test_size", 0.2)
 
+    user_model = params["layers"][0]
```

Comment:
maybe it might be good to validate that for classical ML, there is only one layer in the architecture